Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1148,10 +1148,10 @@ void FrameLoader::resetMultipleFormSubmissionProtection()
 
 void FrameLoader::updateFirstPartyForCookies()
 {
-    if (RefPtr localParent = dynamicDowncast<LocalFrame>(m_frame->tree().parent()))
-        setFirstPartyForCookies(localParent->document()->firstPartyForCookies());
-    else
+    if (bool isMainFrame = !m_frame->tree().parent())
         setFirstPartyForCookies(m_frame->document()->url());
+    else
+        setFirstPartyForCookies(m_frame->page()->mainFrameURL());
```

Comment:
`documentURLDidChange`, which calls `setMainFrameURL`, gets called inside `Document::setURL`

Could we just remove the true case here and always call `setFirstPartyForCookies(m_frame->page()->mainFrameURL());`?